File: src/main/java/com/ocelot/vehicle/jei/plugin/fluidmixer/FluidMixerRecipeMaker.java
package com.ocelot.vehicle.jei.plugin.fluidmixer;

import com.mrcrayfish.vehicle.crafting.FluidExtract;
import com.mrcrayfish.vehicle.crafting.FluidMixerRecipe;
import com.mrcrayfish.vehicle.crafting.FluidMixerRecipes;
import mezz.jei.api.IJeiHelpers;
import mezz.jei.api.recipe.IStackHelper;
import net.minecraft.item.ItemStack;
import net.minecraftforge.fluids.Fluid;
import net.minecraftforge.fluids.FluidStack;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * @author Ocelot
 */
public final class FluidMixerRecipeMaker
{
    private FluidMixerRecipeMaker()
    {
    }

    public static List<FluidMixerRecipeWrapper> getRecipes(IJeiHelpers jeiHelpers)
    {
        IStackHelper stackHelper = jeiHelpers.getStackHelper();
        FluidMixerRecipes fluidMixerRecipes = FluidMixerRecipes.getInstance();
        Map<FluidMixerRecipe, FluidExtract> mixingMap = fluidMixerRecipes.getMixingMap();

        List<FluidMixerRecipeWrapper> recipes = new ArrayList<>();
        mixingMap.forEach((input, output) ->
        {
            List<ItemStack> itemInputs = stackHelper.getSubtypes(input.getIngredient());
            Fluid fluidInput1 = input.getFluids()[0];
            Fluid fluidInput2 = input.getFluids()[1];
            recipes.add(new FluidMixerRecipeWrapper(jeiHelpers.getGuiHelper(), itemInputs, new FluidStack(fluidInput1, input.getFluidAmount(fluidInput1)), new FluidStack(fluidInput2, input.getFluidAmount(fluidInput2)), output.createStack()));
        });
        return recipes;
    }
}

File: src/main/java/com/ocelot/vehicle/jei/plugin/fluidmixer/FluidMixerRecipeMaker.kt
package com.ocelot.vehicle.jei.plugin.fluidmixer

import com.mrcrayfish.vehicle.crafting.FluidMixerRecipes
import mezz.jei.api.IJeiHelpers
import net.minecraftforge.fluids.FluidStack
import java.util.*

object FluidMixerRecipeMaker {

    fun getRecipes(helpers: IJeiHelpers): List<FluidMixerRecipeWrapper> {
        val stackHelper = helpers.stackHelper
        val fluidMixerRecipes = FluidMixerRecipes.getInstance()
        val mixingMap = fluidMixerRecipes.mixingMap

        val recipes = ArrayList<FluidMixerRecipeWrapper>()

        for (entry in mixingMap.entries) {
            val mixingRecipe = entry.key
            val mixingResult = entry.value
            val itemInputs = stackHelper.getSubtypes(mixingRecipe.ingredient)
            val recipe = FluidMixerRecipeWrapper(itemInputs, FluidStack(mixingRecipe.fluids[0], mixingRecipe.amounts[0]), FluidStack(mixingRecipe.fluids[1], mixingRecipe.amounts[1]), mixingResult.createStack())
            recipes.add(recipe)
        }

        return recipes
    }
}